linode/networkreservedips/framework_datasource_list.go

```diff
@@ -0,0 +1,72 @@
+package networkreservedips
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewDataSourceList() datasource.DataSource {
+	return &DataSourceList{
+		BaseDataSource: helper.NewBaseDataSource(
+			helper.BaseDataSourceConfig{
+				Name:   "linode_reserved_ip_list",
+				Schema: &frameworkDataSourceListSchema,
+			},
+		),
+	}
+}
+
+type DataSourceList struct {
+	helper.BaseDataSource
+}
+
+func (d *DataSourceList) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	tflog.Debug(ctx, "Read data.linode_reserved_ip_list")
+
+	var data DataSourceListModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	client := d.Meta.Client
+
+	// List all reserved IPs
+	ips, err := client.ListReservedIPAddresses(ctx, nil)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to list Reserved IP Addresses",
+			err.Error(),
+		)
+		return
+	}
+
+	reservedIPs := make([]ReservedIPObject, len(ips))
+	for i, ip := range ips {
```

Comment:
Is `vpc_nat_1_1` missing?

Reply:
I followed the schema of the closest existing resource - `linode_networking_ip`. We can still add this. Is it always null for reserved IPs? I see that when I reserve an IP address the `vpc_nat_1_1` value is null. Need further clarification.